Added stm32 nucleo_f767zi board support

rolandma opened the pull request:
Tested with Blinky

Commit:
Added stm32 nucleo_f767zi board support

Commit:
Added stm32 nucleo_f767zi board support

Commit:
Added stm32 nucleo_f767zi board support

Commit:
Fixed files with Dos Line Endings

Commit:
Fixed files with Dos Line Endings

Commit:
Fixed files with Dos Line Endings

rolandma committed:
Updated documentation

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Updated documentation

Signed-off-by: Roland Ma <rolandma@yahoo.com>

dbkinder commented on boards/arm/nucleo_f767zi/doc/index.rst:
```diff
@@ -0,0 +1,226 @@
+Overview
+********
+
+The STM32 Nucleo-144 boards offer combinations of performance and power that
```
Hmmm... the title is for this doc is``ST Nucleo F767ZI`` board, but you're documenting a different board here: ``STM32 Nucleo-144``.  Something's amiss.

rolandma replied:
Following several other Nucleo document format for this Nucleo-144 series board, such as nucleo_f746zg and nucleo_f756zg. I personally have no preference.

dbkinder replied:
Well, that's a bit confusing but I get it now.  The STM32 Nucleo-144 boards are a family while NUCLEO-L496ZG and NUCLEO-L496ZG-P, for example, are specific members of that family of boards.

dbkinder commented on boards/arm/nucleo_f767zi/doc/index.rst:
```diff
@@ -0,0 +1,226 @@
+   :height: 720px
+   :alt: Nucleo F767ZI
+
+More information about the board can be found at the `Nucleo F746ZG website`_.
```
I suspect this should be `` at the `Nucleo f767zi website`_.``

rolandma replied:
Updated.

kaidoho commented on dts/arm/st/f7/stm32f767.dtsi:
```diff
@@ -0,0 +1,58 @@
+#include <st/f7/stm32f7.dtsi>
+
+/ {
+	/* 64KB DTCM @ 20000000, 240KB SRAM1 @ 20010000, 16KB SRAM2 @ 2004C000 */
```
This is a copy of the **F746**, but I guess it should be a **F767** 

`/* 128KB DTCM @ 20000000, 368KB SRAM1 @ 20020000, 16KB SRAM2 @ 2007C000 */`

	sram0: memory@20020000 {
		compatible = "mmio-sram";
		reg = <0x20020000 DT_SIZE_K(384)>;
	};

	dtcm: memory@20000000 {
		compatible = "arm,dtcm";
		reg = <0x20000000 DT_SIZE_K(128)>;
	};

kaidoho commented on soc/arm/st_stm32/stm32f7/Kconfig.soc:
```diff
@@ -13,6 +13,9 @@ config SOC_STM32F723XX
 config SOC_STM32F746XX
 	bool "STM32F746XX"
 
+config SOC_STM32F767XX
```
For **F767**, please enable the double precision as it is done for **F769**. Could you keep the way the devices are ordered within this file? F756, F767, F769 etc. 

`select CPU_HAS_FPU_DOUBLE_PRECISION`

kaidoho commented on soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx:
```diff
@@ -0,0 +1,26 @@
+endif # GPIO_STM32
+
+config NUM_IRQS
+	default 98
```
`default 110` for **F767**

rolandma committed:
Updated configuration files.
Fixed checkpatch issues.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Updated configuration files.
Fixed checkpatch issues.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Fixing comment from dbkinder about the website.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Fixing comment from dbkinder about the website.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Updated Kconfig.defconfig.stm32f767xx and Kconfig.soc for kaidoho's comments.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Updated Kconfig.defconfig.stm32f767xx and Kconfig.soc for kaidoho's comments.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma reviewed (commented):
Commit de9821673b:
1: UC3 Title does not follow [subsystem]: [subject] (and should not start with literal subsys:): "Updated Kconfig.defconfig.stm32f767xx and Kconfig.soc for kaidoho's comments."
1: UC5 Title exceeds max length (77>75): "Updated Kconfig.defconfig.stm32f767xx and Kconfig.soc for kaidoho's comments."
1: UC6 Body has no content, should at least have 1 line.

What are the UCx? How to fix them?

rolandma committed:
Fixed GitLint error of trailing white space.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Fixed GitLint error of trailing white space.

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Fixed #UC3 #UC6 Gitlint issues:
1. Fixed #UC3
2. Fixed #UC6
3. Fixed #5f3f478c07698f00db6fbdc1c53bf132689d2d27 author email
4. Fixed #787d59a6be336eeccafc15f5a5145aa932c032cf author email

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Fixed #UC3 #UC6 Gitlint issues:
1. Fixed #UC3
2. Fixed #UC6
3. Fixed #5f3f478c07698f00db6fbdc1c53bf132689d2d27 author email
4. Fixed #787d59a6be336eeccafc15f5a5145aa932c032cf author email

Signed-off-by: Roland Ma <rolandma@yahoo.com>

erwango commented:
> What are the UCx? How to fix them?

These are errors reported on the commit messages, if the errors are not enough explicit, you can check https://docs.zephyrproject.org/latest/contribute/#commit-guidelines. Also, you can check the style of other commit messages in zephyr repo.

erwango reviewed (changes requested):
Some changes. 

One point about the commits:
When the PR is merged, the commits of the PR are put as is on top of zephyr master branch.
So, the PRs should be clean of any errors, we will not merge commits with fixes or fixes of commits. Fixes done during the review should be integrated directly in the commits.
For this change you're supposed to have only 2 commits, in this order:
- One commit for the dts addition
- One commit for the board addition

So, what I advise you to:
- squash all current commits
- split in two commits (soc and boards)
- for the review, use git fixup and autosquash rebase to integrate the review comments.

To use a fixup commit, put the following in the commit message and then commit:
`fixup! SHA1 of the commit to fix` or `fixup! Title of the commit to fix`
You can add as many fixup commits as you want.
Once you're done,  rebase using --autosquash so git integrate the fixups:
`git rebase -i HEAD~X --autosquash`
X should be higher than then number of commits on your branch

rolandma commented on dts/arm/st/f7/stm32f767.dtsi:
```diff
@@ -0,0 +1,58 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/f7/stm32f7.dtsi>
```
Similar as STM32F769.dtsi, the RAM structure is a little different from STM32F746, so I got RAM0 redefined error if I include the stm32f746.dtsi

erwango replied:
Ok, indeed.
We'd need to review the whole stm32f7 series definitions to solve this. Let's keep that aside for now.

dbkinder reviewed (approved):
Thanks for explaining about STM32 Nucleo-144 boards vs. specific instances of a board in this family.

rolandma committed:
Change the copyright messages

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Change the copyright messages

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Change the copyright messages

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
fixup! 77bfc53

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
fixup! f50192c

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
fixup! 895f214

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
fixup! 3874a21

Signed-off-by: Roland Ma <rolandma@yahoo.com>

rolandma committed:
Add stm32 nucleo_f767zi board support

Signed-off-by: Roland Ma <rolandma@yahoo.com>

# Conflicts:
#	boards/arm/nucleo_f767zi/CMakeLists.txt
#	boards/arm/nucleo_f767zi/Kconfig.board
#	boards/arm/nucleo_f767zi/Kconfig.defconfig
#	boards/arm/nucleo_f767zi/arduino_r3_connector.dtsi
#	boards/arm/nucleo_f767zi/board.cmake
#	boards/arm/nucleo_f767zi/doc/index.rst
#	boards/arm/nucleo_f767zi/nucleo_f767zi.dts
#	boards/arm/nucleo_f767zi/nucleo_f767zi.yaml
#	boards/arm/nucleo_f767zi/nucleo_f767zi_defconfig
#	boards/arm/nucleo_f767zi/pinmux.c
#	boards/arm/nucleo_f767zi/support/openocd.cfg
#	dts/arm/st/f7/stm32f767.dtsi
#	dts/arm/st/f7/stm32f767Zi.dtsi
#	soc/arm/st_stm32/stm32f7/Kconfig.defconfig.stm32f767xx

rolandma commented:
Closing this PR because the changes has been re-applied into another PR Add board definition for stm32 nucleo_f767zi #21656 for clean up. All the changes remain the same.
Thanks.
